File: pdsa/week1/Graded-3.py

def odd_one(l):
	a, b = type(l[0]), type(l[1])
	if a == b:
		for el in l[2:]:
			c = type(el)
			if c != a:
				return c.__name__
	else:
		c = type(l[2])
		return a.__name__ if b == c else b.__name__

File: degree-level/c-programming/timed-mock-5/graded-3.c
#include <stdio.h>
#include <string.h>


struct Person {
	char name[100];
	int age;
	char city[100];
};


void findYoungestPersons(struct Person people[], int n) {
	int min_age = people[0].age;
	for (int i = 1; i < n; i++) {
		if (people[i].age < min_age) {
			min_age = people[i].age;
		}
	}
	for (int i = 0; i < n; ++i) {
		if (people[i].age == min_age) {
			printf("%s: %s\n", people[i].name, people[i].city);
		}
	}
}



int main()
{
int n;
scanf("%d", &n);
struct Person persons[n];
for (int i = 0; i < n; i++)
{
scanf("%s",persons[i].name);
scanf("%d",&persons[i].age);
scanf("%s",persons[i].city);
}
findYoungestPersons(persons, n);
return 0;
}